### PR TITLE
Class DieParameters

### PR DESCRIPTION
**Pull request by audrey01mit**
- Created class `DieParameters` (containing `unit_die_size`, `pad_size`, `outline`, `die_layer`, `pads_layer` etc).
- Now every cell takes a DieParameters in argument instead of a repetitive list of params. It makes the code way clearer when calling cells and simpler for contributors.
- Some functions in utility are under this class as they are inherent from the die's parameters.
- Also helped fixing issue #63

### CODE REVIEW

File: src/qnngds/design.py [hunk 1]
```diff
@@ -335,7 +335,9 @@ def __init__(
         self.chip_w = chip_w
         self.chip_margin = chip_margin
         self.N_dies = N_dies
-        self.die_w = die_w
+        self.die_size = unit_die_size
```
Comment (reed-foster) [inline src/qnngds/design.py:338]: do you still need these if they get passed into the die parameters?

**Review by reed-foster — COMMENTED** [review]
Looking good! I didn't manage to get through everything so I'll have a more thorough look tomorrow. I trust Val to review too if I don't get around to it hahaha

File: src/qnngds/cells.py [hunk 2]
```diff
@@ -160,27 +147,25 @@ def vdp(
     for layer in layers_to_probe:
         TEST_LAY = pg.deepcopy(VDP)
         if layer in layers_to_outline:
-            TEST_LAY = pg.outline(TEST_LAY, outline)
+            TEST_LAY = pg.outline(TEST_LAY, die_parameters.outline)
         TEST_LAY.name = f"VDP(lay={layer})"
```
Comment (reed-foster) [inline src/qnngds/cells.py:151]: I might expand `lay` to `layer` just for clarity

File: src/qnngds/design.py (hunk 1 above)
Comment (audrey01mit) [reply src/qnngds/design.py:338]: This class is for any user creating a new design, technically the die_parameters class is to simplify the code for contributors so yes a design should still start by fixing all of these paarmeetrs (without having to create first a die_parameetrs class and then pass it to the design class)

**Pull request merged**